src/list/mod.rs

```diff
@@ -398,6 +398,22 @@ where
     }
 }
 
+impl<T, P> Drop for List<T, P>
```

Comment:
Add a small comment saying we implement this to avoid a stack overflow when dropping lists.